src/Compilers/CSharp/Test/CommandLine/ErrorLoggerTests.cs

```diff
@@ -61,6 +61,13 @@ public static void Main(string[] args)
         }
 
         [Fact]
+        public void GetAssemblyFileVersionOnDevBuild()
+        {
+            var cmd = new MockCSharpCompiler(null, _baseDirectory, new[] { "/nologo", "/preferreduilang:en" });
+            Assert.Equal("42.42.42.42424 (<developer build>)", cmd.GetAssemblyFileVersion());
+        }
```

Comment:
Good suggestion. 
I also found out that the `FileVersionInfo.ProductVersion` API and found that it could return null. Added a few cases.